### ISSUE TITLE
Unexpected brace wrapping for `hx-vals`

### ISSUE THREAD

**Issue by shouya** [bigskysoftware/htmx#2885]
I want to make an element to include dynamic values generated via javascript. The suggested way to do this is through `hx-vals=...` attribute. Here's an example use case where I want to implement a list of toggles whose values are aggregated before sending to the server:

```html
<ul hx-get="..." hx-vals="js:aggregateVals()" hx-trigger="click from:li">
  <li data-enabled="1" data-id="1" onclick="this.dataset.enabled=1-(+this.dataset.enabled)">1</li>
  <li data-enabled="0" data-id="2" onclick="this.dataset.enabled=1-(+this.dataset.enabled)">2</li>
  <li data-enabled="0" data-id="3" onclick="this.dataset.enabled=1-(+this.dataset.enabled)">3</li>
</ul>
<script>
function aggregateVals() {
  const enabledEntries = Array.from(document.querySelectorAll("li")).filter(x => +x.dataset.enabled).map(x => x.dataset.id).join(",");
  if (enabledEntries === "") {
    return {};
  } else {
    return {entries: enabledEntries};
  }
}
</script>
```

My intention here is to skip the query parameter altogether if no entry is enabled.

However, according to the [current implementation](https://github.com/bigskysoftware/htmx/blob/230262cf9347ca2f89703c3348660e487ca02c5b/src/htmx.js#L3741-L3743), `hx-vals` will implicitly wrap the value after `js:`, so the actual evaluated expression becomes `{aggregateVals()}`. I tried to get around the problem with this:

```html
<ul hx-vals="js:...aggregateVals()" ...>
```

It works as expected but the syntax looks very unnatural to me. I suppose the implicit braces may be added to `hx-vals` to be compatible with the now deprecated [`hx-vars`](https://htmx.org/attributes/hx-vars/) attribute. Since this implicit syntax is never mentioned in the documentation of [`hx-vals`](https://htmx.org/attributes/hx-vals/), I suppose such usage is not encouraged. If so can we remove it in exchange for a nicer syntax for the more dynamic use case as I demonstrated?

**Comment by MichaelWest22** [comment]
Yeah I think the {} wrapping has been added to make sure every return is in the right object format to be processed and prevent non object values being provided.  The example documentation gives `hx-vals='js:{myVal: calculateValue()}'` which forces it to provide a value of myVal with a dynamic value but you are wanting the option to not provide the property when it has not been set.  Someone just raised a possible fix to improve hx-vals in #2884 and it seems that an improved version of that fix could make your use case easier as it could be changed to strip undefined or maybe null properties allowing you to simplify your code by having `hx-vals='js:{"entries": getEntries()}'` and getting getEntries to return undefined or a value and it then only sends that parameter when it's valid.

**Comment by shouya** [comment]
I agree with the pros and cons you listed. Indeed, although I think the implicitly addition of brace is sort of gratuitous, I see it does not worth it to break compatibility promise just for the sake of an arguably cleaner syntax. On a hindsight I admit I'm a bit too aggressive in pushing the issue when I had the issue and can only resort to a hackish solution.

I agree with the current way of enforcing the braces. It's a way to suggest the attribute only accepts json/js objects. I see that `js:{ ...aggregateVals() }`  is cleaner and admit that's better than a plain `js:aggregateVals()` for this reason.

Looking back with the root of my confusion is that there seems to be no way of providing dynamical keys to the request params, at least I find nothing in the documentation. So when I searched through the source and found this unintended usage, I have to call it a hack.

So how about we consider this spread operator officially supported? By which I mean adding an example in `hx-vals` documentation for `hx=vals="js:{...generateKeys()}"` and add a test case for it?

**Comment by Telroshan** [comment]
Sounds good to me, dynamic keys is definitely a usecase worth documenting.
A documentation PR + test would be very welcome @shouya if you feel like it!
If so, feel free to either close your existing PR and make a new one, or reuse it & rename + force push!
